include/h2o.h

```diff
@@ -1563,6 +1563,11 @@ typedef struct st_h2o_compress_context_t {
      */
     void (*compress)(struct st_h2o_compress_context_t *self, h2o_iovec_t *inbufs, size_t inbufcnt, h2o_send_state_t state,
                      h2o_iovec_t **outbufs, size_t *outbufcnt);
+    /**
+     * decompress
+     */
+    void (*decompress)(struct st_h2o_compress_context_t *self, h2o_iovec_t *inbufs, size_t inbufcnt, h2o_send_state_t state,
+                     h2o_iovec_t **outbufs, size_t *outbufcnt);
```

Comment:
Compression context is a context for transforming one representation (e.g. a gzip-ed file) to another (e.g. an decompressed file). A _single_ context can only be used for one purpose; it cannot be used for compressing and decompressing.

Therefore, you should not add a `decompress` callback.

You should either:
* create `h2o_compress_context_t` that decompresses the input passed into the `compress` callback
 * you could possibly rename the word _compress_ used in various names, but I am not sure if that is worth the effort, considering the fact that we would decompress gzip only
* create a separate ostream dedicated for decompressing a gzipped content
* implement decompression in lib/handler/file.c

The above list reflects my preference of addressing the issue.